src/Microsoft.OData.Core/UriParser/SemanticAst/SelectExpandClauseExtensions.cs

```diff
@@ -86,7 +86,37 @@ internal static void GetSelectExpandPaths(this SelectExpandClause selectExpandCl
         /// <returns>String list generated from selected items</returns>
         internal static List<string> GetCurrentLevelSelectList(this SelectExpandClause selectExpandClause)
         {
-            return selectExpandClause.SelectedItems.Select(GetSelectString).Where(i => i != null).ToList();
+            List<string> levelSelectList = new List<string>();
+          
+            foreach (var selectItem in selectExpandClause.SelectedItems)
+            {
+                WildcardSelectItem wildcardSelect = selectItem as WildcardSelectItem;
+                if (wildcardSelect != null)

```

Comment:
since you don't actually use wildcardSelectItem (except to check for null) you could just do:

 if (selectItem is WildcardSelectItem)

Reply:
Done. #Resolved